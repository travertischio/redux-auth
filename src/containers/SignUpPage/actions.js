/*
 *
 * SignUpPage actions
 *
 */

import {
  SIGN_UP_ACTION,
  SIGN_UP_SUCCESS_ACTION,
  SIGN_UP_FAILED_ACTION,
} from './constants';

export const signUpAction = (userData) => ({
  type: SIGN_UP_ACTION,
  payload: userData,
});

export const signUpSuccessAction = (response) => ({
  type: SIGN_UP_SUCCESS_ACTION,
  payload: response,
});

export const signUpFailedAction = (response) => ({
  type: SIGN_UP_FAILED_ACTION,
  payload: response,
});
