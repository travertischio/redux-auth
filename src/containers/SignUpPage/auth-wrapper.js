import { UserAuthWrapper } from 'redux-auth-wrapper';
import { routerActions } from 'react-router-redux';
import { selectUser } from '../AuthenticationProvider/selectors';
import config from '../../config';

const SignUpForNotAuthenticatedUsers = UserAuthWrapper({
  authSelector: selectUser,
  predicate: isNotAuthenticated,
  redirectAction: routerActions.replace,
  failureRedirectPath: config.redirectPathAfterSignIn,
  allowRedirectBack: false,
  wrapperDisplayName: 'SignUpForNotAuthenticatedUsers',
});

function isNotAuthenticated(user) {
  return !user;
}

export default SignUpForNotAuthenticatedUsers;
