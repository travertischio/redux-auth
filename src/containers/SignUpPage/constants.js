/*
 *
 * SignUpPage constants
 *
 */

export const SIGN_UP_ACTION = 'app/SignUpPage/SIGN_UP';
export const SIGN_UP_SUCCEED_ACTION = 'app/SignUpPage/SUCCEED';
export const SIGN_UP_FAILED_ACTION = 'app/SignUpPage/FAILED';
