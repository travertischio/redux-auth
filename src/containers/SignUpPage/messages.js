/*
 * SignUpPage Messages
 *
 * This contains all the text for the SignUpPage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  pageTitle: {
    id: 'app.containers.SignUpPage.pageTitle',
    defaultMessage: 'Sign Up',
  },
  pageDescription: {
    id: 'app.containers.SignUpPage.pageDescription',
    defaultMessage: 'Sign Up',
  },
});
