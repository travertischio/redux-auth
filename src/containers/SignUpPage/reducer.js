/*
 *
 * SignUpPage reducer
 *
 */
import { fromJS } from 'immutable';
import {
  SIGN_UP_ACTION,
  SIGN_UP_SUCCEED_ACTION,
  SIGN_UP_FAILED_ACTION,
} from './constants';

const initialState = fromJS({});

function signUpPageReducer(state = initialState, action) {
  switch (action.type) {
    case SIGN_UP_ACTION:
      return onSignUpAction(state);
    case SIGN_UP_SUCCEED_ACTION:
      return onSignUpSucceedAction(state);
    case SIGN_UP_FAILED_ACTION:
      return onSignUpFailedAction(state, action.payload);
    default:
      return state;
  }
}

export const onSignUpAction = (state) => state.merge({
  loading: true,
  errorMessage: false,
});

export const onSignUpSucceedAction = (state) => state.merge({
  loading: false,
  errorMessage: false,
});

// TODO handle field errors!
export const onSignUpFailedAction = (state, rejection) => {
  const { response } = rejection;
  let errorMessage = 'Ooops, something went wrong, please try again later.';

  if (response.status === 400) {
    errorMessage = response.data.non_field_errors;
  }

  return state.merge({
    loading: false,
    errorMessage,
  });
};

export default signUpPageReducer;
