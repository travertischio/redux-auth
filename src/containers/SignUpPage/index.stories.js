import React from 'react';
import PropTypes from 'prop-types';
import { storiesOf } from '@kadira/storybook';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import SignUpPage from './';
import { selectIsAuthenticated } from '../../';

let SignUpPageWrapper = (props) => {
  const { isAuthenticated } = props;

  return (
    <main>
      {!isAuthenticated && <SignUpPage />}
      {isAuthenticated && <h1>You are authenticated!</h1>}
    </main>
  );
};

SignUpPageWrapper.propTypes = {
  isAuthenticated: PropTypes.bool.isRequired,
};

const mapStateToProps = createStructuredSelector({
  isAuthenticated: selectIsAuthenticated,
});

SignUpPageWrapper = connect(mapStateToProps)(SignUpPageWrapper);

storiesOf('Sign Up Page', module)
  .add('Sign Up', () => <SignUpPageWrapper />);
