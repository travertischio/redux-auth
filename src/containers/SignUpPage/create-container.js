import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Helmet from 'react-helmet';
import { injectIntl } from 'react-intl';
import { createStructuredSelector } from 'reselect';
import { compose } from 'recompose';
import selectSignUpPage from './selectors';
import messages from './messages';
import { signUpAction } from './actions';
import config from '../../config';

export default function createSignUpContainer(PageComponent, options = {}) {
  const mapStateToProps = createStructuredSelector({
    SignUpPage: selectSignUpPage,
  });

  const mapDispatchToProps = {
    onSubmitForm: signUpAction,
  };

  @compose(config.signUpAuthWrapper)
  @injectIntl
  @connect(mapStateToProps, mapDispatchToProps)
  class SignUpContainer extends PureComponent { // eslint-disable-line react/prefer-stateless-function
    static propTypes = {
      intl: PropTypes.object.isRequired,
    };

    render() {
      const { formatMessage } = this.props.intl;
      const pageTitle = formatMessage(messages.pageTitle);
      const pageDescription = formatMessage(messages.pageDescription);

      return (
        <div className={options.className || 'sign-up-page'}>
          <Helmet
            title={pageTitle}
            meta={[
              {
                name: 'description',
                content: pageDescription,
              },
            ]}
          />

          <PageComponent {...this.props} />
        </div>
      );
    }
  }

  return SignUpContainer;
}
