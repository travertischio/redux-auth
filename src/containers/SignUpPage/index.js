/*
 *
 * SignUpPage
 *
 */

import { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Helmet from 'react-helmet';
import { injectIntl } from 'react-intl';
import { createStructuredSelector } from 'reselect';
import SignUpForm from '../../components/SignUpForm';
import UserIsNotAuthenticated from '../../hocs/AuthWrappers/UserIsNotAuthenticated';
import makeSelectSignUpPage from './selectors';
import messages from './messages';
import { signUpAction } from './actions';

const mapStateToProps = createStructuredSelector({
  SignUpPage: makeSelectSignUpPage(),
});

const mapDispatchToProps = {
  onSubmitForm: signUpAction,
};

@UserIsNotAuthenticated
@injectIntl
@connect(mapStateToProps, mapDispatchToProps)
export default class SignUpPage extends PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {
      onSubmitForm,
      SignUpPage: {
        loading,
        errorMessage,
      },
    } = this.props;
    const { formatMessage } = this.props.intl;
    const pageTitle = formatMessage(messages.pageTitle);
    const pageDescription = formatMessage(messages.pageDescription);

    return (
      <div>
        <Helmet
          title={pageTitle}
          meta={[
            {
              name: 'description',
              content: pageDescription,
            },
          ]}
        />

        { loading && <div>Processing... Please wait.</div> }
        { errorMessage && <div>{errorMessage}</div> }

        <SignUpForm onSubmit={onSubmitForm} />
      </div>
    );
  }
}

SignUpPage.propTypes = {
  SignUpPage: PropTypes.object,
  onSubmitForm: PropTypes.func,
  intl: PropTypes.object,
};
