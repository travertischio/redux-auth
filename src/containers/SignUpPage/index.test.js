import React from 'react';
import {
  // getStoreWithInitialState,
  // findActionByType,
  // mountWithIntl,
  createComponentWithRouter,
} from 'react-unit-testing-utils';
// import { SubmissionError } from 'redux-form/immutable';
import SignUpPage from './index';
// import { destroyPageAction } from './actions';
// import {
//   // SIGN_UP_ACTION,
//   // DESTROY_PAGE_ACTION,
// } from './constants';

describe('<SignUpPage />', () => {
  // describe('when submitting form', () => {
  //   let wrapper;
  //   const store = getStoreWithInitialState({
  //     auth: {},
  //     signUpPage: {},
  //   });


  //   beforeEach(() => {
  //     wrapper = mountWithIntl(<SignUpPage />, {
  //       store,
  //     });
  //   });

  //   it('should reject SubmissionError when backend rejects request', (done) => {
  //     const errorResponse = {
  //       response: {
  //         data: {
  //           nonFieldErrors: ['Username already exists'],
  //         },
  //       },
  //     };

  //     wrapper
  //       .children()
  //       .get(0) // 1
  //       .props
  //       .onSubmitForm()
  //       .catch((error) => {
  //         expect(error).toEqual(new SubmissionError(errorResponse));
  //         done();
  //       });

  //     const recivedAction = findActionByType(store, SIGN_UP_ACTION);

  //     recivedAction.reject(errorResponse);
  //   });

  //   it('should dispatch action SIGN_UP_ACTION', () => {
  //     const firstName = 'John';
  //     const email = 'tester@test.com';
  //     const password = 'zxy123';

  //     wrapper.find('input[name="firstName"]').simulate('change', { target: { value: firstName } });
  //     wrapper.find('input[name="email"]').simulate('change', { target: { value: email } });
  //     wrapper.find('input[name="password"]').simulate('change', { target: { value: password } });
  //     wrapper.find('input[name="confirmPassword"]').simulate('change', { target: { value: password } });
  //     wrapper.find('form').simulate('submit');

  //     const recivedAction = findActionByType(store, SIGN_UP_ACTION);

  //     expect(recivedAction.type).toEqual(SIGN_UP_ACTION);
  //   });
  // });

  // it('should dispatch action DESTROY_PAGE_ACTION when the component unmount', () => {
  //   const initialState = {
  //     auth: {},
  //     signUpPage: {},
  //   };
  //   const { wrapper, store } = createComponentWithRouter(<SignUpPage />, initialState);
  //   wrapper.unmount();
  //   const recivedAction = findActionByType(store, DESTROY_PAGE_ACTION);
  //   expect(recivedAction).toEqual(destroyPageAction());
  // });

  it('should NOT render SignUpPage when user IS authenticated', () => {
    const initialState = {
      auth: {
        tokenData: {},
        userData: {},
      },
    };
    const { wrapper } = createComponentWithRouter(<SignUpPage />, initialState);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render SignUpPage when user IS NOT authenticated', () => {
    const initialState = {
      auth: {},
      signUpPage: {},
    };
    const { wrapper } = createComponentWithRouter(<SignUpPage />, initialState);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render SignUpPage with loading indicator when loading is in the props', () => {
    const initialState = {
      auth: {},
      signUpPage: {
        loading: true,
      },
    };
    const { wrapper } = createComponentWithRouter(<SignUpPage />, initialState);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render SignUpPage with error message when errorMessage is in the props', () => {
    const initialState = {
      auth: {},
      signUpPage: {
        errorMessage: 'Ooops, something went wrong, please try again later.',
      },
    };
    const { wrapper } = createComponentWithRouter(<SignUpPage />, initialState);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
