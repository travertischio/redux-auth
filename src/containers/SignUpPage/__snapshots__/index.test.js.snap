// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpPage /> should NOT render SignUpPage when user IS authenticated 1`] = `<main />`;

exports[`<SignUpPage /> should render SignUpPage when user IS NOT authenticated 1`] = `
<main>
  <div
    className="sign-up-page"
  >
    <div>
      <form
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className=""
        >
          <div
            className="form-item"
          >
            <label
              htmlFor="firstName"
            >
              Name
            </label>
            <input
              className="textfield"
              id="firstName"
              name="firstName"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder="Name"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className=""
        >
          <div
            className="form-item"
          >
            <label
              htmlFor="email"
            >
              E-mail
            </label>
            <input
              className="textfield"
              id="email"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder="E-mail"
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          className=""
        >
          <div
            className="form-item"
          >
            <label
              htmlFor="password"
            >
              Password
            </label>
            <input
              className="textfield"
              id="password"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          className=""
        >
          <div
            className="form-item"
          >
            <label
              htmlFor="confirmPassword"
            >
              Repeat password
            </label>
            <input
              className="textfield"
              id="confirmPassword"
              name="confirmPassword"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder="Repeat password"
              type="password"
              value=""
            />
          </div>
        </div>
        <div>
          <button
            disabled={true}
            type="submit"
          >
            Sign Up
          </button>
        </div>
      </form>
    </div>
  </div>
</main>
`;

exports[`<SignUpPage /> should render SignUpPage with error message when errorMessage is in the props 1`] = `
<main>
  <div
    className="sign-up-page"
  >
    <div>
      <div>
        Ooops, something went wrong, please try again later.
      </div>
      <form
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className=""
        >
          <div
            className="form-item"
          >
            <label
              htmlFor="firstName"
            >
              Name
            </label>
            <input
              className="textfield"
              id="firstName"
              name="firstName"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder="Name"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className=""
        >
          <div
            className="form-item"
          >
            <label
              htmlFor="email"
            >
              E-mail
            </label>
            <input
              className="textfield"
              id="email"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder="E-mail"
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          className=""
        >
          <div
            className="form-item"
          >
            <label
              htmlFor="password"
            >
              Password
            </label>
            <input
              className="textfield"
              id="password"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          className=""
        >
          <div
            className="form-item"
          >
            <label
              htmlFor="confirmPassword"
            >
              Repeat password
            </label>
            <input
              className="textfield"
              id="confirmPassword"
              name="confirmPassword"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder="Repeat password"
              type="password"
              value=""
            />
          </div>
        </div>
        <div>
          <button
            disabled={true}
            type="submit"
          >
            Sign Up
          </button>
        </div>
      </form>
    </div>
  </div>
</main>
`;

exports[`<SignUpPage /> should render SignUpPage with loading indicator when loading is in the props 1`] = `
<main>
  <div
    className="sign-up-page"
  >
    <div>
      <div>
        Processing... Please wait.
      </div>
      <form
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className=""
        >
          <div
            className="form-item"
          >
            <label
              htmlFor="firstName"
            >
              Name
            </label>
            <input
              className="textfield"
              id="firstName"
              name="firstName"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder="Name"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className=""
        >
          <div
            className="form-item"
          >
            <label
              htmlFor="email"
            >
              E-mail
            </label>
            <input
              className="textfield"
              id="email"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder="E-mail"
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          className=""
        >
          <div
            className="form-item"
          >
            <label
              htmlFor="password"
            >
              Password
            </label>
            <input
              className="textfield"
              id="password"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          className=""
        >
          <div
            className="form-item"
          >
            <label
              htmlFor="confirmPassword"
            >
              Repeat password
            </label>
            <input
              className="textfield"
              id="confirmPassword"
              name="confirmPassword"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              placeholder="Repeat password"
              type="password"
              value=""
            />
          </div>
        </div>
        <div>
          <button
            disabled={true}
            type="submit"
          >
            Sign Up
          </button>
        </div>
      </form>
    </div>
  </div>
</main>
`;
